codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -378,16 +384,34 @@ protected Http2FrameWriter frameWriter() {
      * Writes a {@code RST_STREAM} frame to the remote endpoint and updates the connection state appropriately.
      */
     @Override
-    public ChannelFuture writeRstStream(ChannelHandlerContext ctx, int streamId, long errorCode,
-            ChannelPromise promise) {
-        Http2Stream stream = connection().stream(streamId);
+    public ChannelFuture writeRstStream(final ChannelHandlerContext ctx, int streamId, long errorCode,
+            final ChannelPromise promise) {
+        final Http2Stream stream = connection().stream(streamId);
+        if (stream == null) {
+            // Don't write a RST_STREAM frame if we are not aware of the stream.
+            return promise.setFailure(streamError(streamId, INTERNAL_ERROR,
+                    "Stream %d not found while writing RST_STREAM frame", streamId));
+        }
+        if (stream.isResetSent()) {
+            // Don't write a RST_STREAM frame if we already wrote a RST_STREAM frame
+            return promise.setFailure(streamError(streamId, INTERNAL_ERROR,
```

Comment:
@Scottmitch I don't think the spec limits us to sending only one RST_STREAM frame, so treating this as an error is problematic.